pkg/util/clusterauthorizer/authorizer.go

```diff
@@ -70,29 +64,33 @@ func (a *azRefreshableAuthorizer) NewRefreshableAuthorizerToken(ctx context.Cont
 	return azidext.NewTokenCredentialAdapter(tokenCredential, scopes), nil
 }
 
-func GetTokenCredential(environment *azureclient.AROEnvironment, credentials *Credentials) (azcore.TokenCredential, error) {
-	return azidentity.NewClientSecretCredential(
-		string(credentials.TenantID),
-		string(credentials.ClientID),
-		string(credentials.ClientSecret),
-		environment.ClientSecretCredentialOptions())
-}
+func GetTokenCredential(environment *azureclient.AROEnvironment) (azcore.TokenCredential, error) {
+	credential, err := azidentity.NewDefaultAzureCredential(environment.DefaultAzureCredentialOptions())
 
-// AzCredentials gets Cluster Service Principal credentials from the Kubernetes secrets
-func AzCredentials(ctx context.Context, client client.Client) (*Credentials, error) {
-	clusterSPSecret := &corev1.Secret{}
-	err := client.Get(ctx, types.NamespacedName{Namespace: AzureCredentialSecretNameSpace, Name: AzureCredentialSecretName}, clusterSPSecret)
 	if err != nil {
 		return nil, err
 	}
 
+	return credential, nil
+}
+
+// AzCredentials gets Cluster Service Principal credentials from the Kubernetes Secret. It returns nil if the Secret is not found.
+func AzCredentials(ctx context.Context, kubernetescli kubernetes.Interface) (Credentials, error) {
```

Comment:
You're right, it's not being used anywhere else and we don't need it anymore. I left it in because it seems like it might be useful in the future and we already have tests for it and everything, but I'll remove it if we think it makes sense to.

Reply:
+1 on cleaning it up - it seems duplicate now :)